lib/experiment/local/client.rb

```diff
@@ -34,13 +47,16 @@ def evaluate(user, flag_keys = [])
       flags = @flags_mutex.synchronize do
         @flags
       end
-      return {} if flags.nil?
-
       user_str = user.to_json
+      if flags.nil?
+        @assignment_service&.track(Assignment.new(user, {}))
+        return {}
+      end
```

Comment:
Do other SDKs also track an empty assignment here? This seems superfluous.

Reply:
Yes, the other SDKs follows the implementation [here](https://github.com/amplitude/experiment-jvm-server/blob/beta-refactor/src/main/kotlin/LocalEvaluationClient.kt#L95), where `assignmentResults` may be empty. Should it not be tracked?